keep/api/routes/topology.py

```diff
@@ -41,17 +43,91 @@ def get_topology_data(
     #     )
 
     try:
-        topology_data = get_all_topology_data(
-            tenant_id, provider_id, service_id, environment
+        topology_data = TopologiesService.get_all_topology_data(
+            tenant_id, session, provider_id, service_id, environment, includeEmptyDeps
         )
-        if not includeEmptyDeps:
-            topology_data = [
-                topology for topology in topology_data if topology.dependencies
-            ]
         return topology_data
     except Exception:
         logger.exception("Failed to get topology data")
         raise HTTPException(
             status_code=400,
             detail="Unknown exception when getting topology data, please contact us",
         )
+
+
+@router.get("/applications", description="Get all applications", response_model=List[TopologyApplicationDtoOut])
```

Comment:
Just to align style with other routes
```suggestion
@router.get(
    "/applications", 
    description="Get all applications", 
    response_model=List[TopologyApplicationDtoOut]
)
```